tests/integration/test_gomod_packages.py

```diff
@@ -32,3 +32,73 @@ def test_gomod_vendor_without_flag(test_env):
         )
     else:
         utils.assert_properly_completed_response(completed_response)
+
+
+def test_gomod_vendor_check_fail(test_env):
+    """
+    Validate failing of gomod vendor request with gomod-vendor-check flag and inconsistent vendor.
+
+    Checks:
+    * The request failed with expected error message
+    """
+    env_data = utils.load_test_data("gomod_vendor_check.yaml")["gomod_vendor_check_fail"]
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"], test_env.get("timeout"))
+    initial_response = client.create_new_request(
+        payload={
+            "repo": env_data["repo"],
+            "ref": env_data["ref"],
+            "flags": env_data["flags"],
+            "pkg_managers": env_data["pkg_managers"],
+        },
+    )
+    completed_response = client.wait_for_complete_request(initial_response)
+    assert completed_response.status == 200
+    assert completed_response.data["state"] == "failed"
+    error_msg = (
+        "The content of the vendor directory is not consistent with go.mod. "
+        "Run `go mod vendor` locally to fix this problem. See the logs for more details."
+    )
+    assert error_msg in completed_response.data["state_reason"], (
+        f"#{completed_response.id}: Request failed correctly, but with unexpected message: "
+        f"{completed_response.data['state_reason']}. Expected message was: {error_msg}"
+    )
+
+
+def test_gomod_vendor_check_pass(test_env):
+    """
+    Validate successful completion of gomod vendor request with gomod-vendor-check flag.
```

Comment:
I think this test copies [the common one](https://github.com/release-engineering/cachito/blob/master/tests/integration/test_packages.py#L25). Could you please move the test data there or point me why you choose to have a new one?

Reply:
I'll move the first case to `test_packages.py`, and keep the `empty_vendor` and `no_vendor` test cases in this file.

Reply:
Great, thank you